subsys/mpsl/fem/Kconfig

```diff
@@ -28,11 +28,17 @@ config MPSL_FEM_NCS_SUPPORTED_FEM_USED
 	bool
 	default n
 
+config MPSL_FEM_API_AVAILABLE
```

Comment:
I don't understand why this extra Konfig symbol is needed. Why can't the user simply use `MPSL_FEM`?

Reply:
MPSL_FEM_CHOICE has no `MPSL_FEM_NONE` value, which would select `none` implementation. On the other hand protocols use FEM API regardless if FEM is used (other than `none` variant) or not (`none` used). FEM `none` implementation requires initialization. This was done by `mpsl_init`. However `MPSL_FEM_ONLY` has been added to use FEM without `mpsl_init`. We need to call `mpsl_fem_init` instead to allow `none` implementation to work. This must be done based of some Kconfigs (the proposal is to base it on `MPSL_FEM_API_AVAILABLE`).

The other approach (I think you suggest it, right?) would be to:
Add `MPSL_FEM_NONE` choice to `MPSL_FEM_CHOICE` (default one).
MPSL_FEM=y would be needed by all protocols using the FEM API, would play the role of proposed `MPSL_FEM_API_AVAILABLE` 

However this approach would change the meaning of `MPSL_FEM`. 
@jciupis do you remember any cons against this approach?

@rugeGerritsen which approach do you consider better?

Reply:
It's a complex scenario, so let me explain.

Consider a protocol driver that uses MPSL FEM APIs unconditionally, similarly to [how the 802.15.4 Radio Driver does it](https://github.com/zephyrproject-rtos/hal_nordic/blob/master/drivers/nrf_802154/driver/src/nrf_802154_trx.c#L366). Let's assume that this protocol driver doesn't depend on MPSL itself - it's possible after recent changes in MPSL's Kconfig integration. Now consider what should happen if the application using this protocol driver is configured to not use any FEM software support, because there's simply no physical device connected. In such case `CONFIG_MPSL=n` and `CONFIG_MPSL_FEM=n`. However, the calls to MPSL FEM API are still present in the protocol driver. They should be resolved to the same implementation used when `CONFIG_MPSL=y` and `CONFIG_MPSL_FEM=n`. This additional symbol is needed to detect such a situation and initialize whatever necessary to allow these "empty" calls. Without it, such configuration (`CONFIG_MPSL=n` and `CONFIG_MPSL_FEM=n`) could also describe a blinky application without any radio-related activities, where this initialization would obviously not be needed.

Reply:
@ankuns the decision is explained in detail in a comment to [KRKNWK-15123](https://projecttools.nordicsemi.no/jira/browse/KRKNWK-15123). The solution you described can easily lead to a NULL-pointer dereference. Checking the pointer before referencing it would create unconditional execution overhead when no FEM is used, which is unacceptable.

Reply:
If we go with `MPSL_FEM_NONE` choice approach:

For "blinky" application (no protocol using FEM API) would be to have `MPSL=n` `MPSL_FEM=n`. MPSL_FEM would gate from any FEM initialization.
For apps using indirectly FEM API but not using MPSL would have `MPSL=n` `MPSL_FEM=y` selected choice for MPSL_FEM_CHOICE (by default new `MPSL_FEM_NONE=y`)
For apps using indirectly FEM API and using MPSL (BLE, 802.15.4 mutti-protocol) would have `MPSL=y` `MPSL_FEM=y` selected choice for MPSL_FEM_CHOICE (by default new `MPSL_FEM_NONE=y`)

But as I mentioned earlier that would change the meaning of `MPSL_FEM` Kconfig symbol.

Reply:
@rugeGerritsen following our F2F discussion I cleaned up `MPSL_FEM_API_AVAILABLE` dependencies and added relevant documentation to the user guide.